lib/publiccloud/ec2.pm

```diff
@@ -241,6 +241,36 @@ sub stop_instance
     die("Failed to stop instance $instance_id") unless ($attempts > 0);
 }
 
+sub suspend_instance {
+    my ($self, $instance) = @_;
+    my $instance_id = $instance->instance_id();
+    my $attempts = 60;
+
+    die("Outdated instance object") if ($instance->public_ip ne $self->get_public_ip());
+
+    assert_script_run("aws ec2 stop-instances --instance-ids $instance_id --hibernate", quiet => 1, timeout => 300);
+
+    while ($self->get_state_from_instance($instance) ne 'stopped' && $attempts-- > 0) {
+        sleep 5;
+    }
+    die("Failed to suspend instance $instance_id") unless ($attempts > 0);
+}
+
+sub resume_instance {
+    my ($self, $instance) = @_;
+    my $instance_id = $instance->instance_id();
+    my $attempts = 60;
+
+    die("Outdated instance object") if ($instance->public_ip ne $self->get_public_ip());
+
+    script_run("aws ec2 start-instances --instance-ids $instance_id", quiet => 1, timeout => 300);
+
+    while ($self->get_state_from_instance($instance) ne 'running' && $attempts-- > 0) {
```

Comment:
I would refactor a `wait_for_state` function in this module, since this is being used twice.

You can then pass 'stopped' and 'running' as parameter.

Reply:
Done.